eq_cir_proxy_service/routers/instrument_router.py

```diff
@@ -0,0 +1,47 @@
+"""Module defines the instrument router for handling requests related to instruments in the EQ CIR Proxy Service."""
+
+from uuid import UUID
+
+from fastapi import APIRouter, HTTPException, Path, Query
+
+from eq_cir_proxy_service.config.logging_config import logging
+from eq_cir_proxy_service.exception import exception_messages
+from eq_cir_proxy_service.services.schema import schema_retrieval_service
+from eq_cir_proxy_service.services.validators.request_validator import (
+    validate_version,
+)
+from eq_cir_proxy_service.types.custom_types import Schema
+
+router = APIRouter()
+logger = logging.getLogger(__name__)
+INSTRUMENT_ID_PATH = Path(..., description="UUIDv4 of the instrument")
+
+
+@router.get("/instrument/{instrument_id}")
+async def get_instrument_by_uuid(
+    instrument_id: UUID = INSTRUMENT_ID_PATH,
+    version: str = Query(default=None, description="Optional version of the instrument"),
+) -> Schema:
+    """Retrieve an instrument by its UUID and optional version."""
+    logger.info("Receiving the instrument id...")
```

Comment:
Might be overkill having both info and debug level logging here? Does the instrument_id need to be in the debug level since it will be in the url anyway? Might be easier just to have the value at the info level?

Reply:
I thought generally that logging values in info is bad practice, as you could end up logging something disclosive.
I know in this case the value is in the url anyway so there is no danger, but prefer to be careful.